### PR TITLE
Support for ARM SVE and better performance for NEON

### PR DESCRIPTION
**Pull request by fwessels**

### CODE REVIEW

**Comment by klauspost** [comment]
Wow, getting SVE support is sluggish: https://go-review.googlesource.com/c/go/+/365034 🐌 

How is the output generated? It is pretty much non-maintainable without the generator(s) but I realize the lack of SVE support probably means there are some external programs needed.

**Comment by fwessels** [comment]
> Wow, getting SVE support is sluggish: https://go-review.googlesource.com/c/go/+/365034 🐌

Yes, have seen that too -- SVE is a quite extensive and rich instruction set and with predicate support etc. it is far from trivial (and a huge effort) to define an alternative syntax that covers essentially everything. So not surprised that it's been going on for a while now.

File: _gen/gen.go [hunk 1]
```diff
@@ -413,7 +454,10 @@ func genMulAvx2(name string, inputs int, outputs int, xor bool) {
 	for _, ptr := range inPtrs {
 		ADDQ(offset, ptr)
 	}
-	// Offset no longer needed unless not regdst
+	// Offset no longer needed unless not regDst
+	if !regDst {
+		SHRQ(U8(3), offset) // divide by 8 since we'll be scaling it up when loading or storing
+	}
```
Comment (klauspost) [inline _gen/gen.go:460]: Can we avoid this? Some Intel react badly to scaling in address resolution, so I tend to avoid it if it is feasible.

**Review by klauspost — COMMENTED** [review]
Just one minor thing. Otherwise LGTM.

**Review by klauspost — APPROVED** [review]

**Pull request merged**